controllers/volumereplication_controller.go

```diff
@@ -179,6 +179,15 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return reconcile.Result{}, err
 	}
 
+	// empty `Status.State` conveys that this is the first reconcile, so we enable replication.
+	if instance.Status.State == "" {
```

Comment:
lets not depend on the state. Even if the state is secondary, send a request for enabling,Demote and Resync. as spec need to be idempotent  it should return success

Reply:
so enable replication on very reconcile?

Reply:
if we can depend on the status This is fine, if we don't want to depend on the status we won't be able to reconcile if someone disabled mirroring in backend or something changed in the backend.

Reply:
ok. makes sense to not depend on status.state.  Removed it.